pkg/azure/actuator.go

```diff
@@ -58,28 +62,58 @@ func (a *Actuator) IsValidMode() error {
 	return errors.New("invalid mode")
 }
 
+func isAzureCredentials(providerSpec *runtime.RawExtension) (bool, error) {
+	codec, err := minterv1.NewCodec()
+	if err != nil {
+		return false, err
+	}
+	unknown := runtime.Unknown{}
+	err = codec.DecodeProviderSpec(providerSpec, &unknown)
+	if err != nil {
+		return false, err
+	}
+	isAzure := unknown.Kind == reflect.TypeOf(minterv1.AzureProviderSpec{}).Name()
+	if !isAzure {
+		log.WithField("kind", unknown.Kind).
+			Info("actuator handles only azure credentials")
+	}
+	return isAzure, nil
+}
+
 func (a *Actuator) Create(ctx context.Context, cr *minterv1.CredentialsRequest) error {
+	if isAzure, err := isAzureCredentials(cr.Spec.ProviderSpec); !isAzure {
+		return err
+	}
```

Comment:
If it is not azure credential, we return the error.
if it managed to decode but found a different Kind than expected, we return nil and the operation is a NoOp.
if it failed to decode (only if the schemes are wrong) we just return the error, which gets handled/logged at a higher level